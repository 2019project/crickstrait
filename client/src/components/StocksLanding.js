import React, { Component } from "react";
//[Nikhil] imports for range slider
import "rc-slider/assets/index.css";
import "rc-tooltip/assets/bootstrap.css";
import ReactDOM from "react-dom";
import Tooltip from "rc-tooltip";
import Slider from "rc-slider";
// import RadarSlider from "../charts/graph";

import {
  getCompany,
  getSectors,
  getIndustries,
  getGainersLosers,
  //[NIKHIL] SCREENER ACTIONS
  getScreenerSearch
} from "../actions/Stocks";
import Script from "react-load-script";
import "../styles/StocksLanding.css";
import "../styles/Admin.css";
import companylogo from "./Common/stockslogo.PNG";
import editlogo from "./Common/edit.png";
import InfiniteScroll from "react-infinite-scroll-component";
import { connect } from "react-redux";
import Loader from "react-loader-spinner";
import Table from "../components/Common/TickerTable";
var slider;

//[Nikhil] rc-slider
//package documentation link
//https://www.npmjs.com/package/rc-slider

//source code http://react-component.github.io/slider/examples/handle.html
const createSliderWithTooltip = Slider.createSliderWithTooltip;
const Range = createSliderWithTooltip(Slider.Range);

export class StocksLanding extends Component {
  state = {
    data: [],
    items: 20,
    loadingstate: false,
    height: 800,
    pageStocks: [],
    loading: false,
    sector: "Basic Materials",
    industry: "Building Materials",
    gainersClick: true,

    //[Nikhil] default slider value
    dividend_value1: 50,
    dividend_value2: 80,
    market_cap_value1: 60,
    market_cap_value2: 70
  };

  //for slider handle cange
  onSliderChange = e => {
    console.log(e);
    this.props.getScreenerSearch(e[0], e[1]);

    //changing the state with the  value selected
    this.setState({ dividend_value1: e[0] });
    this.setState({ dividend_value2: e[1] });
  };
  onSliderChange2 = e => {
    this.props.getScreenerSearch(e[0], e[1]);

    this.setState({ market_cap_value1: e[0] });
    this.setState({ market_cap_value2: e[1] });
  };

  componentDidMount() {
    //for loading graph.js
    const script = document.createElement("script");
    script.src = "C:Users\nikhiDesktopgraphgraph.js";
    script.async = true;
    script.onload = () => this.scriptLoaded();

    document.body.appendChild(script);

    this.props.getCompany(this.state.industry); //getting all the companies based on an industry selected
    this.props.getSectors(); //getting all the sectors
    this.props.getIndustries(this.state.sector); //getting all the industries based on a sector selected
    this.props.getGainersLosers(this.state.sector); //getting all the gainers and losers based on a sector selected
  }

  componentWillReceiveProps(nextProps) {
    nextProps.stocks.length > 0
      ? this.displayCompanies(nextProps.stocks)
      : console.log(0, " Stocks");
  }

  OnSelectSector = e => {
    this.setState({
      sector: e.target.value //setting state for the sector
    });
    this.props.getIndustries(e.target.value); //getting all the industries based on a sector selected
    this.props.getGainersLosers(e.target.value); //getting all the gainers and losers based on a sector selected
  };

  OnSelectIndustry = e => {
    this.setState({
      industry: e.target.value //setting state for the industry
    });
    this.props.getCompany(e.target.value); //getting all the companies based on an industry selected
  };

  displayCompanies = Stocks => {
    console.log(Stocks);
    const { items } = this.state;

    if (Stocks.length === 0) return;
    let pageStocks = [];
    for (let i = 0; i < items; i++) {
      pageStocks.push(Stocks[i]);
    }
    this.setState({ pageStocks });
    console.log("PageStocks array", this.state.pageStocks.length);
  };

  loadMoreItems = () => {
    setTimeout(() => {
      this.setState({
        items: this.state.items + 10
      });
      this.displayCompanies(this.props.stocks);
    }, 1000);
  };

  render() {
    return (
      <div>
        {/* real code starts here  */}
        <div>
          <div id="stocks-landing-page-slider">
            {/* CLLING THE COMPONENT WITH THE RS SLIDER PACKAGE  */}
            <Range
              // SETTING THE MINIMUM VALUE
              min={0}
              // SETTING THE MAXIMUM VALUE
              max={1000}
              //SETTING THE DEFAULT VALUE WHICH IS DEFINED IN THE STATE OF THE COMPONENT
              defaultValue={[
                this.state.dividend_value1,
                this.state.dividend_value2
              ]}
              //ON CHANGING CALLING THE SLIDERCHANGE
              onChange={this.onSliderChange}
            />
          </div>
          <p>Value: {this.state.dividend_value1}</p>
          <p>Value: {this.state.dividend_value2}</p>
        </div>
        market cap
        <div>
          <div id="stocks-landing-page-slider">
            {/* CLLING THE COMPONENT WITH THE RS SLIDER PACKAGE  */}
            <Range
              // SETTING THE MINIMUM VALUE
              min={0}
              // SETTING THE MAXIMUM VALUE
              max={1000}
              //SETTING THE DEFAULT VALUE WHICH IS DEFINED IN THE STATE OF THE COMPONENT
              defaultValue={[
                this.state.market_cap_value1,
                this.state.market_cap_value2
              ]}
              //ON CHANGING CALLING THE SLIDERCHANGE
              onChange={this.onSliderChange2}
              name={"slider1"}
            />
          </div>
          <p>Value: {this.state.market_cap_value1}</p>
          <p>Value: {this.state.market_cap_value2}</p>
        </div>
        {/* ends */}
        {this.props.gainersLosers["0"] ? (
          <div id="stocks_main_container">
            {/* <p>STOCKS</p> */}
            <div id="stocks_filter">
              <select
                type="text"
                id="stocks_dropdown_sectors"
                className="stocks_dropdown"
                name="sector"
                value={this.state.sector}
                onChange={this.OnSelectSector}
              >
                {this.props.sectors.map((sectors, index) => (
                  <>
                    <option name="choice" id={"sector" + index}>
                      {sectors}
                    </option>
                    {/*mapping all the sectors to select from the dropdown*/}
                  </>
                ))}
              </select>
              <select
                type="text"
                className="stocks_dropdown"
                id="stocks_dropdown_industries"
                name="industries"
                value={this.state.industry} //changing the value of industry when selected
                onChange={this.OnSelectIndustry} //on change perform this function
              >
                <option name="choice">Select an Industry</option>
                {this.props.industries.map((industries, index) => (
                  <>
                    <option name="choice" id={"industry" + index}>
                      {industries}
                    </option>
                    {/* mapping all the industries in the dropdown based on the selected sector*/}
                  </>
                ))}
              </select>
            </div>

            {/* <div class="slidecontainer"> */}

            {/* </div> */}

            <div id="stocks_main_grid_container">
              {this.props.isLoading ? (
                <div id="stocks_loader">
                  <Loader type={Loader} color="#2c3e50" textAlign="center" />
                </div>
              ) : (
                <InfiniteScroll
                  dataLength={this.state.pageStocks.length} //This is important field to render the next data
                  next={this.loadMoreItems}
                  hasMore={true}
                  height={600}
                  loader={
                    <div id="stocks_loader">
                      <Loader
                        type={Loader}
                        color="#2c3e50"
                        textAlign="center"
                      />
                    </div>
                  }
                  endMessage={
                    <p style={{ textAlign: "center" }}>
                      <b>Yay! You have seen all the stocks</b>
                    </p>
                  }
                >
                  <div id="stocks_grid_container">
                    {this.state.pageStocks.map((stocks, index) =>
                      stocks ? (
                        <div id="stocks_main_grid_details">
                          {/* -------------------------- */}
                          {/* <div class="box"> */}
                          <a href="#editpopup">
                            <img id="stocks_edit" alt="edit" src={editlogo} />
                          </a>
                          {/* </div> */}
                          <div id="editpopup" class="admin_overlay">
                            <div class="admin_popup">
                              <h2>Edit Stock</h2>
                              <a class="admin_close" href="#">
                                &times;
                              </a>
                              <div class="admin_content">
                                <button>Upload</button>
                                <button>Edit</button>
                              </div>
                            </div>
                          </div>
                          {/* -------------------------- */}
                          <div
                            id="stocks_grid_details"
                            onClick={() => {
                              this.props.history.push(
                                "/companydetail/" + stocks.ticker_id, //pushing to the company details page with ticker id of a stock when that particular stock card is clicked
                                { stocks }
                              );
                            }}
                          >
                            <img
                              id="stocks_img"
                              alt="logo"
                              src={
                                stocks.ticker_logo == null
                                  ? companylogo
                                  : "data:image/jpeg;base64," +
                                    stocks.ticker_logo
                              }
                            />
                            <div id="stocks_ticker">
                              {stocks["ticker_name"]}
                            </div>
                            {/* mapping the ticker name from the api*/}
                            <div id="stocks_name">{}</div>
                            <div id="stocks_flex_details_one">
                              <div
                                id="stocks_closed_price"
                                className="stocks_details_title"
                              >
                                Closed Price:
                              </div>
                              <div id="stocks_details">
                                {stocks["Share Price"]}USD
                              </div>
                              {/* mapping the share price from the api */}
                            </div>
                            <div id="stocks_flex_details_two">
                              <div
                                id="stocks_market_cap"
                                className="stocks_details_title"
                              >
                                Market Cap:
                              </div>
                              <div id="stocks_details">
                                {stocks["MarketCap"]}M
                              </div>
                              {/* mapping the market cap from the api */}
                            </div>
                          </div>
                        </div>
                      ) : null
                    )}
                  </div>
                </InfiniteScroll>
              )}
            </div>
            <div id="stocks_table">
              <div id="stocks_div_buttons">
                <button
                  id="button_stocks_gainers"
                  className={
                    this.state.gainersClick === true //changing the color of gainers button when clicked on it
                      ? "stocks_gainers"
                      : "stocks_losers"
                  }
                  onClick={() => this.setState({ gainersClick: true })} //on clicking the gainers button, the state of gainersClick is set true
                >
                  Gainers
                </button>
                <button
                  id="button_stocks_losers"
                  className={
                    this.state.gainersClick === false //changing the color of losers button when clicked on it
                      ? "stocks_gainers"
                      : "stocks_losers"
                  }
                  onClick={() => this.setState({ gainersClick: false })} //on clicking the losers button, the state of gainersClick is set false
                >
                  Losers
                </button>
              </div>
              <Table
                tableHeaders={[
                  "Ticker",
                  "Chng (%)",
                  "Market Cap",
                  "Share Price"
                ]}
                tableData={
                  this.state.gainersClick === true //displaying the gainers data in the table if state of gainersClick is true that is when gainers button is clicked else losers data is displayed
                    ? this.props.gainersLosers["0"].gainers
                    : this.props.gainersLosers["0"].losers
                }
              />
            </div>
          </div>
        ) : (
          <div id="stocks_main_loader">
            <Loader type={Loader} color="#2c3e50" textAlign="center" />
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  stocks: state.stocksReducer.stocks,
  sectors: state.stocksReducer.sectors,
  industries: state.stocksReducer.industries,
  gainersLosers: state.stocksReducer.gainersLosers,
  isLoading: state.LoadingReducer.isLoading
});

export default connect(mapStateToProps, {
  getCompany,
  getSectors,
  getIndustries,
  getGainersLosers,
  getScreenerSearch
})(StocksLanding);
