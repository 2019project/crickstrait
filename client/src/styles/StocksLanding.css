@import url("https://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.css");
@import url("https://code.jquery.com/jquery-1.11.3.min.js");
@import url("https://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.js");
* {
  font-family: Karla;
}

/* //for screener [nikhil] */

/* .slidecontainer {
  width: 100%;
}

.slider {
  -webkit-appearance: none;
  width: 100%;
  height: 25px;
  background: #d3d3d3;
  outline: none;
  opacity: 0.7;
  -webkit-transition: 0.2s;
  transition: opacity 0.2s;
}

.slider:hover {
  opacity: 1;
}

.slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  appearance: none;
  width: 25px;
  height: 25px;
  background: #4caf50;
  cursor: pointer;
}

.slider::-moz-range-thumb {
  width: 25px;
  height: 25px;
  background: #4caf50;
  cursor: pointer;
} */

#stocks_main_container {
  width: 100%;
  margin: 20px 0px 0px 15px;
}

#stocks_main_container p {
  font-size: 34px;
  font-weight: 600;
  margin: 0px 0px 10px 5px;
}

#stocks_main_grid_container {
  float: left;
  width: 70%;
}

#stocks_grid_container {
  margin: 2px 5px 0px 5px;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 15px;
}

#stocks_main_grid_details {
  border: 1px solid #cacaca;
  border-radius: 4px;
  background-color: #ffffff;
  text-align: right;
}

#stocks_grid_details {
  background-color: #ffffff;
  /* border: 1px solid #cacaca;
  border-radius: 4px; */
  padding: 10px;
  text-align: center;
  cursor: pointer;
}

#stocks_main_grid_details:hover {
  box-shadow: 0px 3px 3px 3px #f5f5f5;
  /* transform: scale(1.02); */
  /* transition: transform 0.3s; */
  transition: all 0.3s ease-out;
}

#stocks_img {
  width: 70px;
}

#stocks_edit {
  width: 22px;
}

#stocks_ticker {
  font-size: 30px;
  font-weight: 400;
}

#stocks_name {
  font-size: 20px;
  color: #707070;
}

#stocks_flex_details_one {
  display: flex;
  justify-content: center;
}

#stocks_flex_details_two {
  display: flex;
  justify-content: center;
}

.stocks_details_title {
  color: #707070;
  margin-right: 5px;
}

#stocks_details {
  color: black;
  font-size: 16px;
}

#stocks_table {
  margin: 0px 40px 0px 0px;
  float: right;
  width: 25%;
  padding: 0px 0px 5px 0px;
  background-color: #ffffff;
  box-shadow: 0px 4px 6px 2px #cacaca;
  border-radius: 4px;
}

#stocks_div_buttons {
  padding: 10px 35px 10px 35px;
  display: flex;
  justify-content: space-between;
}

.stocks_gainers {
  width: 140px;
  color: #ffffff;
  background-color: #39abf7;
  border: none;
  border-radius: 4px;
  font-size: 18px;
  padding: 2px;
  cursor: pointer;
}

.stocks_losers {
  width: 140px;
  color: #707070;
  background-color: #ffffff;
  border: none;
  border-radius: 4px;
  font-size: 18px;
  padding: 2px;
  cursor: pointer;
}

.stocks_losers:hover {
  color: #39abf7;
}

#stocks_table_title {
  display: flex;
  justify-content: space-around;
  background-color: #2c3e50;
}

#stocks_table_title p {
  padding: 8px;
  margin: 0px;
  font-size: 16px;
  font-weight: 600;
  color: #ffffff;
}

#stocks_grid_table {
  width: 100%;
  border-collapse: collapse;
}

#stocks_grid_table th {
  background-color: #2c3e50;
  color: #fff;
  padding: 10px;
}

#stocks_grid_table td {
  padding: 15px;
  font-size: 15 px;
}

#positiveChange {
  color: #27ae60;
}

#negativeChange {
  color: #ff4d4d;
}
/* #stocks_grid_table_details {
  text-align: center;
  padding: 10px;
  margin: 0px;
  font-size: 16px;
} */

#stocks_div_filter {
  border: 1px solid black;
}

#stocks_filter {
  margin: 30px 0px 30px 5px;
  display: flex;
}

.stocks_dropdown {
  background-color: #ffffff;
  color: #39abf7;
  border: 1.5px solid #39abf7;
  height: 30px;
  border-radius: 4px;
  margin: 0px 30px 0px 0px;
}

#stocks_loader {
  margin: 50px 0px 50px 500px;
}

#stocks_main_loader {
  margin: 260px 0px 0px 680px;
}
/* ------------------------------------------------------------------------------ */

@media only screen and (max-width: 1224px) {
  #stocks_grid_container {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media only screen and (max-width: 920px) {
  #stocks_grid_container {
    grid-template-columns: 1fr 1fr 1fr;
  }
  #stocks_main_container {
    margin: 0px 0px 0px 0px;
  }
  #stocks_main_grid_container {
    width: 100%;
  }
  #stocks_table {
    display: none;
  }
  #stocks_loader {
    margin: 30px 0px 30px 350px;
  }
  #stocks_main_loader {
    margin: 165px 0px 0px 325px;
  }
}

@media only screen and (max-width: 755px) {
  #stocks_grid_container {
    grid-template-columns: 1fr 1fr 1fr;
  }
  #stocks_main_container {
    margin: 0px 0px 0px 0px;
  }
  #stocks_main_grid_container {
    width: 100%;
  }
  #stocks_table {
    display: none;
  }
}

@media only screen and (max-width: 466px) {
  #stocks_grid_container {
    grid-template-columns: 1fr;
  }
  #stocks_table {
    display: none;
  }
}

@media only screen and (max-width: 426px) {
  #stocks_grid_container {
    margin: 0px 10px 0px 10px;
    grid-template-columns: 1fr;
  }
  #stocks_flex_details_two {
    display: none;
  }
  #stocks_img {
    float: left;
  }
  #stocks_main_container {
    margin: 0px 0px 0px 0px;
  }
  #stocks_main_grid_container {
    width: 100%;
  }
  #stocks_table {
    display: none;
  }
  #stocks_filter {
    margin: 30px 0px 30px 5px;
    display: flex;
  }
  #stocks_loader {
    margin: 20px 0px 20px 125px;
  }
  #stocks_main_loader {
    margin: 205px 0px 0px 120px;
  }
  #stocks_filter {
    margin: 20px 0px 15px 5px;
    display: block;
    text-align: center;
  }
  .stocks_dropdown {
    margin: 0px 0px 5px 0px;
  }
}
