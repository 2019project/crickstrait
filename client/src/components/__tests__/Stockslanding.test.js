import React from "react";
import { StocksLanding } from "../StocksLanding";
import {
  getCompany,
  getSectors,
  getIndustries,
  getGainersLosers
} from "../../actions/Stocks";
import { shallow } from "enzyme";

const stocks = [
  {
    ticker_id: 169,
    ticker_name: "ANTM",
    ticker_logo: "image",
    MarketCap: "21223.9902",
    "Share Price": "51.35"
  }
];
const gainersLosers = [
  {
    isIndex: false,
    gainers: [
      {
        tickerValues: {
          change_percent: "+30.37",
          "Market Cap": "68.00M",
          "Share Price": "4.38"
        },
        ticker_name: "HPJ",
        ticker_id: 1030
      }
    ],
    losers: [
      {
        tickerValues: {
          change_percent: "-33.33",
          "Market Cap": "13.00M",
          "Share Price": "0.03"
        },
        ticker_name: "TGLO",
        ticker_id: 2007
      }
    ]
  }
];
const sectors = ["Technology", "Healthcare"];
const industries = ["Compauter H/w", "online Media"];

const wrapper = shallow(
  <StocksLanding
    stocks={stocks}
    gainersLosers={gainersLosers}
    sectors={sectors}
    industries={industries}
    getCompany={getCompany}
    getSectors={getSectors}
    getIndustries={getIndustries}
    getGainersLosers={getGainersLosers}
  />
);

describe("Testing Stocks Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should check for component did mount as a function", () => {
    const componentDidMount = jest.spyOn(
      StocksLanding.prototype,
      "componentDidMount"
    );
    wrapper.instance().componentDidMount();
    expect(componentDidMount).toHaveBeenCalled();
  });
  it("Table should have button with text Gainers", () => {
    expect(wrapper.find("#button_stocks_gainers").text()).toBe("Gainers");
  });
  it("Table should have button with text Losers", () => {
    expect(wrapper.find("#button_stocks_losers").text()).toBe("Losers");
  });
  it("should simulate button click on Gainers tab ", () => {
    wrapper.find("#button_stocks_gainers").simulate("click");
    expect(wrapper.state().gainersClick).toBe(true);
  });
  it("should simulate button click on Gainers tab ", () => {
    wrapper.find("#button_stocks_losers").simulate("click");
    expect(wrapper.state().gainersClick).toBe(false);
  });
  it("should simulate change on selector dropdown ", () => {
    wrapper.find("#stocks_dropdown_sectors").simulate("change");
  });
  it("should simulate click on stocks card ", () => {
    wrapper.find("#stocks_grid_details0").simulate("click");
  });
  it("Company logo should be shown in each stocks card", () => {
    expect(wrapper.find("#stocks_img").text()).toBe("image");
  });
  it("Ticker Name should be shown in each stocks card", () => {
    expect(wrapper.find("#stocks_ticker").props().children).toBe("ANTM");
  });
  it("Closed price should be shown in each stocks card", () => {
    expect(wrapper.find("#stocks_closed_price").text()).toBe("51.35");
  });
  it("Market cap should be shown in each stocks card", () => {
    expect(wrapper.find("stocks_market_cap").text).toBe("21223.9902");
  });
  it("should check for presense of OnSelectSector function", () => {
    let OnSelectSector = jest.spyOn(wrapper.instance(), "OnSelectSector");
    expect(OnSelectSector).toBeTruthy();
  });
  it("should check for presense of OnSelectSector function", () => {
    let OnSelectSector = jest.spyOn(wrapper.instance(), "OnSelectSector");
    expect(OnSelectSector).toBeTruthy();
  });
  it("should check for presense of OnSelectIndustry function", () => {
    let OnSelectIndustry = jest.spyOn(wrapper.instance(), "OnSelectIndustry");
    expect(OnSelectIndustry).toBeTruthy();
  });
  it("should check for presense of displayCompanies function", () => {
    let displayCompanies = jest.spyOn(wrapper.instance(), "displayCompanies");
    expect(displayCompanies).toBeTruthy();
  });
  it("should check for presense of loadMoreItems function", () => {
    let loadMoreItems = jest.spyOn(wrapper.instance(), "loadMoreItems");
    expect(loadMoreItems).toBeTruthy();
  });

  it("Ticker Name should be shown in each stocks card", () => {
    expect(wrapper.find("#industry0").props().children).toBe(industries[0]);
  });

  it("Ticker Name should be shown in each stocks card", () => {
    expect(wrapper.find("#sectors0").props().children).toBe(sectors[0]);
  });

  it("should have ticker_name click on stocks card ", () => {
    expect(wrapper.find("#stocks_ticker").props().children).toBe(sectors[0]);
  });
});
