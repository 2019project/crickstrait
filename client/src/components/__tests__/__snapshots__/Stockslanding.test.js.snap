// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Stocks Component should mount the component 1`] = `
<div>
  <Radar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgb(57,171,247)",
            "borderColor": "rgb(44,62,80)",
            "data": Array [
              65,
              75,
              70,
              80,
              60,
              80,
            ],
            "fill": true,
            "pointBackgroundColor": "orange",
            "pointBorderColor": "rgba(200,0,0,0.6)",
            "pointBorderWidth": 3,
            "pointHoverRadius": 10,
            "pointRadius": 6,
            "radius": 6,
          },
        ],
        "labels": Array [
          "Market Cap",
          "Dividend",
          "PE Ratio",
          "Debt to Equity Ratio",
          "Share Price",
        ],
      }
    }
    options={
      Object {
        "legend": Object {
          "position": "left",
        },
        "scale": Object {
          "pointLabels": Object {
            "fontSize": 18,
          },
          "ticks": Object {
            "beginAtZero": true,
            "max": 100,
            "min": 0,
            "stepSize": 20,
          },
        },
      }
    }
  />
  <div
    id="stocks_main_container"
  >
    <div
      id="stocks_filter"
    >
      <select
        className="stocks_dropdown"
        id="stocks_dropdown_sectors"
        name="sector"
        onChange={[Function]}
        type="text"
        value="Basic Materials"
      >
        <option
          id="sector0"
          name="choice"
        >
          Technology
        </option>
        <option
          id="sector1"
          name="choice"
        >
          Healthcare
        </option>
      </select>
      <select
        className="stocks_dropdown"
        id="stocks_dropdown_industries"
        name="industries"
        onChange={[Function]}
        type="text"
        value="Building Materials"
      >
        <option
          name="choice"
        >
          Select an Industry
        </option>
        <option
          id="industry0"
          name="choice"
        >
          Compauter H/w
        </option>
        <option
          id="industry1"
          name="choice"
        >
          online Media
        </option>
      </select>
      <label
        for="toggle-1"
        id="advance-label"
      >
        Advance Search
      </label>
      <input
        id="toggle-1"
        type="checkbox"
      />
      <div
        id="stocks-advance-filter-grid-main-container"
      >
        <div
          id="advance-fliter-dividend"
        >
          <div
            id="advance-filter-title"
          >
            <p
              id="advance-filter-dividend-p"
            >
              Dividend 
            </p>
          </div>
          <div
            id="stocks-landing-page-slider"
          >
            <ComponentWrapper
              defaultValue={
                Array [
                  -50,
                  80,
                ]
              }
              handleStyle={
                Array [
                  Object {},
                ]
              }
              max={1000}
              min={-50}
              onChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
            />
          </div>
          <div
            id="advance-filter-debt-to-equity-ratio-value"
          >
            <p
              id="advance-filter-value"
            >
              -50
            </p>
            <p
              id="advance-filter-value2"
            >
              1000
            </p>
          </div>
        </div>
        <div
          id="advance-filter-market-cap"
        >
          <div
            id="advance-filter-title"
          >
            <p
              id="advance-filter-market-cap-p"
            >
              market cap
            </p>
          </div>
          <div
            id="stocks-landing-page-slider"
          >
            <ComponentWrapper
              defaultValue={
                Array [
                  60,
                  70,
                ]
              }
              handleStyle={
                Array [
                  Object {},
                ]
              }
              max={1000}
              min={0}
              name="slider1"
              onChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
            />
          </div>
          <div
            id="advance-filter-debt-to-equity-ratio-value"
          >
            <p
              id="advance-filter-value"
            >
              0
            </p>
            <p
              id="advance-filter-value2"
            >
              100
            </p>
          </div>
        </div>
        <div
          id="advance-filter-share-price"
        >
          <div
            id="advance-filter-title"
          >
            <p
              id="advance-filter-share-price-p"
            >
              share price 
            </p>
          </div>
          <div
            id="stocks-landing-page-slide"
          >
            <ComponentWrapper
              defaultValue={
                Array [
                  43,
                  72,
                ]
              }
              handleStyle={
                Array [
                  Object {},
                ]
              }
              max={1000}
              min={0}
              onChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
            />
          </div>
          <div
            id="advance-filter-debt-to-equity-ratio-value"
          >
            <p
              id="advance-filter-value"
            >
              0
            </p>
            <p
              id="advance-filter-value2"
            >
              100
            </p>
          </div>
        </div>
        <div
          id="advance-filter-price-to-equity"
        >
          <div
            id="advance-filter-title"
          >
            <p
              id="advance-filter-price-to-equity-p"
            >
              PRICE TO EQUITY
               
            </p>
          </div>
          <div
            id="stocks-landing-page-slider"
          >
            <ComponentWrapper
              defaultValue={
                Array [
                  89,
                  44,
                ]
              }
              handleStyle={
                Array [
                  Object {},
                ]
              }
              max={1000}
              min={0}
              onChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
            />
          </div>
          <div
            id="advance-filter-debt-to-equity-ratio-value"
          >
            <p
              id="advance-filter-value"
            >
              0
            </p>
            <p
              id="advance-filter-value2"
            >
              100
            </p>
          </div>
        </div>
        <div
          id="advance-filter-debt-to-equity"
        >
          <div
            id="advance-filter-title"
          >
            <p
              id="advance-filter-debt-to-equity-p"
            >
              DEBT TO EQUITY RATIO
            </p>
          </div>
          <div
            id="stocks-landing-page-slider"
          >
            <ComponentWrapper
              defaultValue={
                Array [
                  45,
                  34,
                ]
              }
              handleStyle={
                Array [
                  Object {},
                ]
              }
              max={1000}
              min={0}
              onChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
            />
          </div>
          <div
            id="advance-filter-debt-to-equity-ratio-value"
          >
            <p
              id="advance-filter-value"
            >
              0
            </p>
            <p
              id="advance-filter-value2"
            >
              100
            </p>
          </div>
        </div>
        <button
          id="advance-filter-button"
          onClick={[Function]}
          type="submit"
        >
          Filter
           
        </button>
      </div>
    </div>
    <div
      id="stocks_main_grid_container"
    >
      <InfiniteScroll
        dataLength={0}
        endMessage={
          <p
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <b>
              Yay! You have seen all the stocks
            </b>
          </p>
        }
        hasMore={true}
        height={600}
        loader={
          <div
            id="stocks_loader"
          >
            <Loader
              className=""
              color="#2c3e50"
              style={Object {}}
              textAlign="center"
              timeout={0}
              type={[Function]}
              visible={true}
            />
          </div>
        }
        next={[Function]}
      >
        <div
          id="stocks_grid_container"
        />
      </InfiniteScroll>
    </div>
    <div
      id="stocks_table"
    >
      <div
        id="stocks_div_buttons"
      >
        <button
          className="stocks_gainers"
          id="button_stocks_gainers"
          onClick={[Function]}
        >
          Gainers
        </button>
        <button
          className="stocks_losers"
          id="button_stocks_losers"
          onClick={[Function]}
        >
          Losers
        </button>
      </div>
      <Connect(TickerTable)
        tableData={
          Array [
            Object {
              "tickerValues": Object {
                "Market Cap": "68.00M",
                "Share Price": "4.38",
                "change_percent": "+30.37",
              },
              "ticker_id": 1030,
              "ticker_name": "HPJ",
            },
          ]
        }
        tableHeaders={
          Array [
            "Ticker",
            "Chng (%)",
            "Market Cap",
            "Share Price",
          ]
        }
      />
    </div>
  </div>
</div>
`;
